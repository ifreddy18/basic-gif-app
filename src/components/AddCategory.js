import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleOnChange = (e) => {
        setInputValue(e.target.value);
    };

    const handleSubmit = (e) => {
        // eslint-disable-next-line no-unused-expressions
        e.preventDefault(); // Previene la recarga del navegador
        
        if ( inputValue.trim().length > 2) {
            setCategories( categories => [inputValue, ...categories]);
            setInputValue('');
        }
    };

    return (
        <form onSubmit={ handleSubmit }>
            <p>{ inputValue }</p>
            <input 
                type="text"
                value={ inputValue }
                onChange={ handleOnChange }
            />
            {/* <button type="submit">Add category</button> */}
        </form>
    );
};

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};