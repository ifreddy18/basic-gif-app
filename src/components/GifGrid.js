import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

    // const [images, setImages] = useState([]);

    const { data:images, loading } = useFetchGifs( category );

    // Hook que se dispara la primera vez y luego no se dispara hasta que el
    // valor de 'category' cambia
    // useEffect(() => {
    //     getFetchGifs( category ).then( setImages );        
    // }, [ category ]); 
    
    return (
        <>
            <h2>{ category }</h2>
               
            { 
                loading && 
                <p className="animate__animated animate__flash">
                    Cargando...
                </p> }

            <div className="card-grid">
                {
                    images.map( img => (
                        <GifGridItem 
                            key={ img.id }
                            { ...img }
                        />
                    ))
                }
            </div>
        </>
    );

};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
};