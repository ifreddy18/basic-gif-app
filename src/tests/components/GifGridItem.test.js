import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem'; 

describe('Tests in component GifGridItem', () => {

    const title = 'This is a title';
    const url = 'http://someUrl.com';
    const wrapper = shallow(
        <GifGridItem title={ title } url={ url } />
    );

    test('Snapshot correcto', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe tener un parrafo con el title', () => {

        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    
    });

    test('Debe tener la imagen igual al url y alt igual al title', () => {

        const img = wrapper.find('img');
        
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    
    });

    test('Debe tener la clase animate__fadeIn', () => {
        
        const div = wrapper.find('div');
        expect( div.hasClass('animate__fadeIn') ).toBe( true );
        
    })
    


});