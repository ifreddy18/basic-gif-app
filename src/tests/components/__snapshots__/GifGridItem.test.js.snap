// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in component GifGridItem Snapshot correcto 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="This is a title"
    src="http://someUrl.com"
  />
  <p>
    This is a title
  </p>
</div>
`;
